package Presence;

message mac_address {
    required bytes addr = 1;
}

message MacAddress {
        required bytes addr = 1;
}

//this message contains client_state information

message presence_client_state {
    enum client_state {
        passerby = 0;
        visitor = 1;
        engaged = 2;
    };

    optional bool is_label = 2;
    optional uint64 label_id = 3;
    optional uint64 label_category_id = 4;
    optional client_state state = 5;
    optional bool is_associated = 6;
}



//The message will have client state information generated by presence context

message pa_client_state{
    repeated presence_client_state presence_client_state = 1;
}


//this message contains proximity event, which is the strongest rssi record reported for a client of all the aps that heard the client.
message pa_proximity_event{
    repeated proximity proximity = 1;
}

message proximity {
   optional string device_id = 1;
   optional mac_address sta_eth_mac = 2;
   optional mac_address radio_mac = 3;
   optional int32 rssi_val = 4;
   optional uint32 noise_floor = 5;
   optional bool associated = 6;
   optional mac_address ap_eth_mac = 7;
}

//this message contains rssi event for a client

message pa_rssi_event
{
    repeated rssi rssi = 1;
}

message rssi {
    optional mac_address sta_eth_mac = 1;
    optional mac_address radio_mac = 2;
    optional int32 rssi_val = 3;
    optional uint32 noise_floor = 4;
    optional bool associated = 5;
    optional bool is_ap = 6;
    optional uint32 age = 7;
    optional string device_id = 8;
   optional mac_address ap_eth_mac = 9;
}

//this message contains presence events paylod
message presence_event {
    enum presence_event_type {
        client_state = 0;
        rssi = 1;
        proximity = 2;
    }
    required uint64 timestamp = 1;
    required string customer_id = 2;
    required string event = 3; // 'presence' category
    required presence_event_type event_type = 4;  // filter event
    optional pa_client_state pa_client_state = 5;
    optional pa_rssi_event pa_rssi_event = 6;
    optional pa_proximity_event pa_proximity_event = 7;
}


